app/controllers/services_controller.rb

```diff
@@ -54,6 +54,24 @@ def show
     render json: ServicesWithResourcePresenter.present(service)
   end
 
+  def init_pdf_crowd_client
+    Pdfcrowd::HtmlToPdfClient.new(Rails.configuration.x.pdfcrowd.username, Rails.configuration.x.pdfcrowd.api_key)
+  end
```

Comment:
I think this type of stuff is normally supposed to go into `config/initializers` so that you only do the initialization once on Rails startup, not for each request. Could you create a `pdfcrowd.rb` file with this in it? I think `algoliasearch.rb` and `sentry.rb` are good examples to borrow from.

Reply:
I am having some trouble setting this one up. It seems like our other initializers like `algoliasearch.rb` and `sentry.rb` are setting config values for their respective module; whereas this line inits a new `Pdfcrowd::HtmlToPdfClient` class instance.. Is there some way I can then access this class instance in the `services_controller`? 🤔

Reply:
Sorry for the delayed reply, but does it not "just work" to instantiate the client in the initializer file? Were you hitting some kind of error when doing that?

Reply:
TIL that you have to assign the class instantiation to a capitalized var to make it globally accessible in Rails. I have a fix for this in a future PR. https://stackoverflow.com/questions/23588674/rails-how-to-set-global-instance-variables-in-initializers

Reply:
Ah, I see. You could also probably create a `module` (really just functioning as a namespace) and then an instance variable within that module's object (similar to a static variable in Java or a class variable in Python). See https://stackoverflow.com/a/8784166 for an example.

The minor benefit of creating an instance variable with access methods on a module vs. just assigning the thing to the capitalized initial letter variable is that the latter is actually syntax for a constant, and constants can never be reassigned to. Modules, on the other hand, can be reopened and their contents modified, which I think would be a better fit for this, since it would simplify cases such as unit testing where we may want to change the value.